### ISSUE TITLE
Excludes for departments?

### ISSUE THREAD

**Issue by thomthom** [rubocop-hq/rubocop#5678]
I was trying to set up `Exclude` rules per department, like this:

```yml
Style:
  Exclude:
    - 'profiling/**/*'
    - 'tests/**/*'
```

However, from what I can gather, `Exclude` only work with `AllCops` or specific cops. Is this correct observation, or is it simply not supported what I'm trying to do?

When I run `rubocop --list-target-files --only Style` it lists files in the directories I tried to exclude.

(What I'm actually trying to do isn't to disable built-in departments, but to exclude custom departments from a RuboCop extension I'm making.)

--------

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.53.0 (using Parser 2.5.0.3, running on ruby 2.4.3 i386-mingw32)
```

**Comment by thomthom** [comment]
Looks like `rubocop-rspec` is trying to do the same thing, but resorting to a workaround:

https://github.com/rubocop-rspec/rubocop-rspec/blob/e51fc4ab47bc4c4a545ee94fec3f4fdfe8e6c05f/lib/rubocop/cop/rspec/cop.rb#L26-L40

Might it be good to add support for department wide excludes in RuboCop? Useful for extensions which features specialized cops that's relevant for only a sub-set of your project.

**Comment by bbatsov** [comment]
There hasn't been much activity on this ticket and our Core Team is spread too thin on so many tasks, so I'll just close it for the sake of having a cleaner lists of tasks to focus on. We'd gladly review a PR, but it's unlikely that'd we're going to tackle this ourselves in the foreseeable future.

**Comment by thomthom** [comment]
I might give a try when I have some pare time. I already have something like this as an option for my custom departments.

Any particular design requirement you can think of? Or is it ok to just model it after Exclude on cops?

**Comment by splattael** [comment]
:wave: from https://github.com/rubocop/rubocop/issues/11148#issuecomment-1336165343

It seems that department excludes works in a limited (cop's `Include` is not overridden) way. They don't work for `--list-target-files` and `--show-cops` though.
